draft-ietf-anima-brski-prm.md

```diff
@@ -1103,589 +1166,583 @@ pledgeenrollrequesttrigger = {
+The Registrar-Agent obtains this artifact within the same TLS session.
+In case the TLS session to the registrar is already closed, the Registrar-Agent establishes a new session as described in {{pvr}}.
+The CA certificates do not need to be correlated to a specific voucher or Enroll-Response; they only need to be fresh.
 
-{{exchangesfig_uc2_5}} shows the request and provisioning of CA certificates in the infrastructure. 
-The following subsections describe the corresponding artifacts. 
+{{exchangesfig_uc2_5}} shows the acquisition of the CA certificate(s) and the following subsections describe the corresponding artifact.
 
 ~~~~ aasvg
 +--------+    +------------+    +-----------+    +--------+    +------+
-| Pledge |    | Registrar- |    |  Domain   |    | Domain |    | MASA |
-|        |    |   Agent    |    | Registrar |    |   CA   |    |      |
+| Pledge |    | Registrar- |    |  Domain   |    |  Key   |    | MASA |
+|        |    |   Agent    |    | Registrar |    | Infra. |    |      |
 +--------+    +------------+    +-----------+    +--------+    +------+
  |                  |                 |                 |   Internet |
  ~                  ~                 ~                 ~            ~
-(5) Request CA Certificates
+(5) Obtain CA Certificates
  ~                  ~                 ~                 ~            ~
  |                  |                 |                 |            |
  |                  |<----(mTLS)----->|                 |            |
- |                  |---cACert-Req--->|                 |            |
- |                  |<--cACert-Resp---|                 |            |
+ |                  |<----caCerts-----|                 |            |
  |                  |                 |                 |            |
  ~                  ~                 ~                 ~            ~
 ~~~~
 {: #exchangesfig_uc2_5 title="CA certificates retrieval exchange" artwork-align="center"}
 
-In case the TLS connection to the registrar is already closed, the Registrar-Agent opens a new TLS connection with the registrar as stated in {{pvr}}.
+As a third step of the interaction with the domain registrar, the Registrar-Agent SHALL obtain the CA-Certificates artificat from the registrar via HTTP-over-TLS GET to the registrar endpoint at `/.well-known/brski/wrappedcacerts`.
+In the request header, the Accept field SHOULD be set to `application/jose+json`.
 
+Upon receiving a GET request at `/.well-known/brski/wrappedcacerts`, the domain registrar MUST reply with the CA-Certificates artifact as defined in {{cacerts-artifact}} in the body of a 200 OK response.
+In the response header, the Content-Type field MUST be set to `application/jose+json`.
 
-### Request Artifact: cACert-Request (cACert-Req)
+TODO(from where do the CA certificates come? I would expect an exchange with the Key Infrastructure, as there was no requirement in Section 6.1 to configure the CA certs at the registrar; also need to be up-to-date)
```

Comment:
I would expect another interaction between registrar and Key Infrastructure for this.

Reply:
The Registrar in BRSKI is also the EST server for the enrollment. This acts as the RA and thus already has the keys available. 
Thinking about this functionality split, the more general term "Key Infrastructure" does hide this split somehow. It is described in RFC 8995 and referred to in Figure 2 of BRSKI-PRM by the statement "e.g. , PKI CA", (as in RFC 8995)

Reply:
"e.g., PKI CA" is an example given how the Key Infra could be implemented.

In this case, I am rather missing the statement in Section 6.1 on what must be provisioned on the Registrar. I would add a normative statement there that it MUST possess the CA certificates, as it acts as EST server.

Reply:
Yes, agree. I thought that this requirement is already there. At least that is also a condition for RFC 8995

Reply:
Clarified from my understanding and could be closed.